pkg/server/cluster_server.go

```diff
@@ -46,16 +65,84 @@ type clusterServer struct {
 func NewClusterServer(kubeConfig, apiserverURL string) (Server, error) {
 	restConfig, err := getClientConfig(kubeConfig)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to create Kubernetes rest client: %v", err)
+		return nil, errors.Wrapf(err, "Failed to create Kubernetes rest client")
 	}
 
-	mc := v1.NewForConfigOrDie(restConfig)
+	mc, err := v1.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating mc client")
+	}
+	oc, err := oconfigv1.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating config client")
+	}
 	return &clusterServer{
 		machineClient:  mc,
+		configClient:   *oc,
 		kubeconfigFunc: func() ([]byte, []byte, error) { return kubeconfigFromSecret(bootstrapTokenDir, apiserverURL) },
 	}, nil
 }
 
+// remoteAddrIsProvisioned returns true if the given IP address responds on a known port.
+// If that's true, the machine is already up and should not be served Ignition.
+func remoteAddrIsProvisioned(remoteAddr string) string {
```

Comment:
hmm, this returns `true` as string or bool ??

Reply:
Neither, it returns the port that responded as a string, or `""` if none.  See what the calling code does.

Reply:
Maybe it'd be better to have it return `error`?  Dunno.

Reply:
I realize now what you meant, the documentation string was wrong.  Fixed that and also changed it to return `error` and clarified the docs!